### PR TITLE
Use vlan filter in getNodeInfoForAutoReg and create/update person even i...

### PR DESCRIPTION
**Pull request by fdurand**
## Description

Added vlan filter in getNodeInfoForAutoReg (set a role to be able to test it in getNormalVlan)
Create or update a person even if the deviced has been registered (Like we did machine auth and after user auth but the pid didn´t changed) 
## Impacts

Autoregistration workflow.
## Delete branch after merge

YES

### CODE REVIEW

File: lib/pf/authentication.pm [hunk 1]
```diff
@@ -452,7 +452,7 @@ If action is set, it will return the value of the action immediately.
 =cut
 
 sub match {
-    my ($source_id, $params, $action) = @_;
+    my ($source_id, $params, $action,$source_ref) = @_;
```
Comment (jrouzierinverse) [inline lib/pf/authentication.pm:456]: Document the new parameter pf::authentication::match

File: lib/pf/vlan.pm [hunk 2]
```diff
@@ -416,10 +417,25 @@ sub getNormalVlan {
                     'unregdate' => $value,
                     'category' => $role,
                     'autoreg' => 'yes',
+                    'pid' => $user_name,
                 );
                 if (defined $role) {
                     %info = (%info, (category => $role));
                 }
+                require pf::person;
+                require pf::lookup::person;
```
Comment (jrouzierinverse) [inline lib/pf/vlan.pm]: Load these modules in the top of file
use pf::person;
use pf::lookup::person;

**Comment by fdurand** [comment]
Can it be merge ?

**Comment by julsemaan** [comment]
Looks good to me

**Pull request merged**